app/controllers/saved_scenario_history_controller.rb

```diff
@@ -72,14 +72,27 @@ def restore
     )
 
     if result.successful?
+      version_tags_result = ApiScenario::VersionTags::FetchAll.call(
```

Comment:
I understand why you did this. But I would like to find a solution that avoids another few API calls to the engine, which I have the feeling are unnecessary – all information should be available in either the controller, or already rendered. What do you think?